contracts/factories/BridgeFactory.sol

```diff
@@ -0,0 +1,274 @@
+    address, // _from 
+    uint256 _id, 
+    uint256 _amount, 
+    bytes memory // _data
+  )
+    public override returns(bytes4)
+  {
+    if (msg.sender == address(arcadeumCoin)) {
+      // Do nothing, just store the ARC until withdrawl
+      require(_id == arcadeumCoinID, "BridgeFactory#onERC1155Received: INVALID_ARC_ID");
+
+    } else if (msg.sender == address(skyweaverAssets)) {
+      // Burn asset received
+      skyweaverAssets.burn(_id, _amount);
+
+    } else {
+      revert("BridgeFactory#onERC1155Received: INVALID_TOKEN");
+    }
+
+    return IERC1155TokenReceiver.onERC1155Received.selector;
+  }
+
+  /**
+   * @notice Burns SW assets when received, or store ARC
+   * @dev Make sure to send the right assets and the correct amount 
+   *      as it's not validated in this contract.
+   * @param _ids     An array containing ids of each Token being transferred
+   * @param _amounts An array containing amounts of each Token being transferred
+   */
+  function onERC1155BatchReceived(
+    address, // _operator
+    address, // _from
+    uint256[] memory _ids,
+    uint256[] memory _amounts,
+    bytes memory // _data
+  )
+    public override returns(bytes4)
+  {
+    if (msg.sender == address(skyweaverAssets)) {
+      // Burn assets received
+      skyweaverAssets.batchBurn(_ids, _amounts);
+
+    // Arc tribute
+    } else if (msg.sender == address(arcadeumCoin)) {
+      // Do nothing, just store the ARC until withdrawl
+      require(_ids[0] == arcadeumCoinID, "BridgeFactory#onERC1155BatchReceived: INVALID_ARC_ID");
+      require(_ids.length == 1, "BridgeFactory#onERC1155BatchReceived: INVALID_ARRAY_LENGTH");
+
+    } else {
+      revert("BridgeFactory#onERC1155BatchReceived: INVALID_TOKEN");
+    }
+
+    return IERC1155TokenReceiver.onERC1155BatchReceived.selector;
+  }
+
+
+  /***********************************|
+  |         Minting Functions         |
+  |__________________________________*/
+
+  /**
+   * @notice Will mint tokens to user
+   * @dev Can only mint up to the dailyLimit in a given day
+   * @param _to      The address that receives the assets
+   * @param _ids     Array of Tokens ID that are minted
+   * @param _amounts Amount of Tokens id minted for each corresponding Token id in _tokenIds
+   */
+  function batchMint(address _to, uint256[] calldata _ids, uint256[] calldata _amounts)
+    external onlyOwnerTier(1)
+  {
+    // Get current period and current available supply
+    uint256 live_period = livePeriod();
+    uint256 available_supply;
+
+    // Get the available supply based on period
+    if (live_period == period) {
+      available_supply = availableSupply;
+    } else {
+      available_supply = periodMintLimit;
+      period = live_period;
+    }
+
+    // Reduce supply based on amount being minted
+    // Will revert if available supply is insufficient
+    for (uint256 i = 0; i < _ids.length; i++) {
+      available_supply = available_supply.sub(_amounts[i]);
+    }
```

Comment:
Inlining the SafeMath here could reduce gas costs, I am not sure if the optimiser does it

Reply:
I would assume it does, but i'll try.

Reply:
Tried and it saves about 10 gas per id, so not going to bother just so the code looks prettier :upside_down_face: